oeis2-1.0.7 fails to build with aeson-2

```
src/Math/OEIS/Internal.hs:124:27: error:
    • Couldn't match expected type ‘Key’ with actual type ‘T.Text’
    • In the first argument of ‘key’, namely ‘k’
      In the first argument of ‘(.)’, namely ‘key k’
      In the second argument of ‘(^?)’, namely ‘key k . _Integer’
    |
124 |   = let d = result ^? key k ._Integer
    |                           ^

src/Math/OEIS/Internal.hs:136:27: error:
    • Couldn't match expected type ‘Key’ with actual type ‘T.Text’
    • In the first argument of ‘key’, namely ‘k’
      In the first argument of ‘(.)’, namely ‘key k’
      In the second argument of ‘(^?)’, namely ‘key k . _String’
    |
136 |   = let d = result ^? key k ._String
    |                           ^

src/Math/OEIS/Internal.hs:150:29: error:
    • Couldn't match expected type ‘Key’ with actual type ‘T.Text’
    • In the first argument of ‘key’, namely ‘k’
      In the first argument of ‘(.)’, namely ‘key k’
      In the second argument of ‘(^?)’, namely ‘key k . _Array’
    |
150 |   = let ds  = result ^? key k . _Array
    |                             ^

src/Math/OEIS/Internal.hs:154:41: error:
    • Couldn't match expected type ‘Key’ with actual type ‘T.Text’
    • In the first argument of ‘key’, namely ‘k’
      In the first argument of ‘(.)’, namely ‘key k’
      In the second argument of ‘(^?!)’, namely ‘key k . nth i . _String’
    |
154 |         let ts  = (\i -> result ^?! key k . nth i . _String) <$> [0..(len - 1)]
    |                                         ^
```

As a Hackage trustee I have revised v1.0.7: https://hackage.haskell.org/package/oeis2-1.0.7/revisions/

---

**EDIT:** Related: https://github.com/commercialhaskell/stackage/issues/6498

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -73,17 +73,17 @@ jobs:
 
     steps:
       - name: Checkout
-        uses: actions/checkout@v3
+        uses: actions/checkout@v4
 
       - name: Setup
-        uses: haskell/actions/setup@v2
+        uses: haskell-actions/setup@v2
         with:
           ghc-version: ${{ matrix.plan.ghc }}
           enable-stack: true
           stack-version: 'latest'
 
       - name: Cache
-        uses: actions/cache@v3
+        uses: actions/cache@v4
         with:
           path: ~/.stack
           key: ${{ runner.os }}-stack-${{ hashFiles('stack.yaml') }}-${{ matrix.plan.resolver }}

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -52,12 +52,14 @@ jobs:
             resolver: lts-18.28
           - ghc: 9.0.2
             resolver: lts-19.0
+          - ghc: 9.2.7
+            resolver: lts-20.18
           - ghc: latest
             resolver: lts
           # Now, 'ghc: head' is unable
           # See: https://github.com/haskell/actions/issues/93
           # - ghc: head
-          - ghc: 9.2.2
+          - ghc: 9.4.5
             resolver: nightly
 
     runs-on: ubuntu-latest

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -54,13 +54,17 @@ jobs:
             resolver: lts-19.0
           - ghc: 9.2.7
             resolver: lts-20.18
+          - ghc: 9.4.8
+            resolver: lts-21.25
+          - ghc: 9.6.5
+            resolver: lts-22.25
+          - ghc: 9.8.2
+            resolver: nightly
           - ghc: latest
             resolver: lts
           # Now, 'ghc: head' is unable
           # See: https://github.com/haskell/actions/issues/93
           # - ghc: head
-          - ghc: 9.4.5
-            resolver: nightly
 
     runs-on: ubuntu-latest
     continue-on-error: ${{ matrix.plan.allow-fail == true }}
@@ -86,7 +90,7 @@ jobs:
 
       - name: Versions
         run: |
-          ghc --version    
+          ghc --version
           stack --version
 
       - name: Configure

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -32,28 +32,6 @@ jobs:
       fail-fast: false
       matrix:
         plan:
-          - ghc: 8.0.2
-            resolver: lts-9.21
-          - ghc: 8.2.2
-            resolver: lts-11.22
-          - ghc: 8.4.4
-            resolver: lts-12.26
-          - ghc: 8.6.4
-            resolver: lts-13.19
-          - ghc: 8.6.5
-            resolver: lts-14.27
-          - ghc: 8.8.2
-            resolver: lts-15.3
-          - ghc: 8.8.4
-            resolver: lts-16.31
-          - ghc: 8.10.3
-            resolver: lts-17.2
-          - ghc: 8.10.7
-            resolver: lts-18.28
-          - ghc: 9.0.2
-            resolver: lts-19.0
-          - ghc: 9.2.7
-            resolver: lts-20.18
           - ghc: 9.4.8
             resolver: lts-21.25
           - ghc: 9.6.5

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -69,17 +69,17 @@ jobs:
 
     steps:
       - name: Checkout
-        uses: actions/checkout@v2
+        uses: actions/checkout@v3
 
       - name: Setup
-        uses: haskell/actions/setup@v1
+        uses: haskell/actions/setup@v2
         with:
           ghc-version: ${{ matrix.plan.ghc }}
           enable-stack: true
           stack-version: 'latest'
 
       - name: Cache
-        uses: actions/cache@v2
+        uses: actions/cache@v3
         with:
           path: ~/.stack
           key: ${{ runner.os }}-stack-${{ hashFiles('stack.yaml') }}-${{ matrix.plan.resolver }}

--- a/oeis2.cabal
+++ b/oeis2.cabal
@@ -36,12 +36,12 @@ library
   hs-source-dirs:
       src
   build-depends:
-      aeson >=1.1 && <2.1
+      aeson >=2 && <2.1
     , base >=4.7 && <5
     , containers >=0.5 && <0.7
     , http-conduit >=2.2 && <2.4
     , lens >=4.15 && <6
-    , lens-aeson >=1.0 && <2
+    , lens-aeson >=1.2 && <2
     , text ==1.2.*
     , vector ==0.12.*
   default-language: Haskell2010
@@ -56,13 +56,13 @@ test-suite oeis2-test
   ghc-options: -threaded -rtsopts -with-rtsopts=-N
   build-depends:
       QuickCheck
-    , aeson >=1.1 && <2.1
+    , aeson >=2 && <2.1
     , base >=4.7 && <5
     , containers >=0.5 && <0.7
     , hspec
     , http-conduit >=2.2 && <2.4
     , lens >=4.15 && <6
-    , lens-aeson >=1.0 && <2
+    , lens-aeson >=1.2 && <2
     , oeis2
     , text ==1.2.*
     , vector ==0.12.*

--- a/oeis2.cabal
+++ b/oeis2.cabal
@@ -1,10 +1,8 @@
 cabal-version: 1.12
 
--- This file has been generated from package.yaml by hpack version 0.34.4.
+-- This file has been generated from package.yaml by hpack version 0.35.1.
 --
 -- see: https://github.com/sol/hpack
---
--- hash: d12ab6c84c2584d3b6f205f5db01501658692c007dbabf678896c297b60ea3bc
 
 name:           oeis2
 version:        1.0.7

--- a/oeis2.cabal
+++ b/oeis2.cabal
@@ -1,11 +1,11 @@
 cabal-version: 1.12
 
--- This file has been generated from package.yaml by hpack version 0.35.1.
+-- This file has been generated from package.yaml by hpack version 0.36.0.
 --
 -- see: https://github.com/sol/hpack
 
 name:           oeis2
-version:        1.0.7
+version:        1.0.8
 synopsis:       Interface for Online Encyclopedia of Integer Sequences (OEIS).
 description:    Release notes are here https://github.com/23prime/oeis2/releases
 category:       Math
@@ -34,14 +34,14 @@ library
   hs-source-dirs:
       src
   build-depends:
-      aeson >=2 && <2.1
-    , base >=4.7 && <5
+      aeson ==2.*
+    , base >=4.17 && <5
     , containers >=0.5 && <0.7
-    , http-conduit >=2.2 && <2.4
+    , http-conduit >=2.2 && <3
     , lens >=4.15 && <6
     , lens-aeson >=1.2 && <2
-    , text ==1.2.*
-    , vector ==0.12.*
+    , text ==2.*
+    , vector ==0.13.*
   default-language: Haskell2010
 
 test-suite oeis2-test
@@ -54,14 +54,14 @@ test-suite oeis2-test
   ghc-options: -threaded -rtsopts -with-rtsopts=-N
   build-depends:
       QuickCheck
-    , aeson >=2 && <2.1
-    , base >=4.7 && <5
+    , aeson ==2.*
+    , base >=4.17 && <5
     , containers >=0.5 && <0.7
     , hspec
-    , http-conduit >=2.2 && <2.4
+    , http-conduit >=2.2 && <3
     , lens >=4.15 && <6
     , lens-aeson >=1.2 && <2
     , oeis2
-    , text ==1.2.*
-    , vector ==0.12.*
+    , text ==2.*
+    , vector ==0.13.*
   default-language: Haskell2010

--- a/package.yaml
+++ b/package.yaml
@@ -18,10 +18,10 @@ dependencies:
 - base         >= 4.7  && < 5
 - text         >= 1.2  && < 1.3
 - vector       >= 0.12 && < 0.13
-- aeson        >= 1.1  && < 2.1
+- aeson        >= 2    && < 2.1
 - containers   >= 0.5  && < 0.7
 - lens         >= 4.15 && < 6
-- lens-aeson   >= 1.0  && < 2
+- lens-aeson   >= 1.2  && < 2
 - http-conduit >= 2.2  && < 2.4
 
 library:

--- a/package.yaml
+++ b/package.yaml
@@ -1,5 +1,5 @@
 name:                oeis2
-version:             1.0.7
+version:             1.0.8
 github:              "23prime/oeis2"
 license:             BSD3
 author:              "Taisuke Hikawa"
@@ -15,14 +15,14 @@ category:            Math
 description:         Release notes are here https://github.com/23prime/oeis2/releases
 
 dependencies:
-- base         >= 4.7  && < 5
-- text         >= 1.2  && < 1.3
-- vector       >= 0.12 && < 0.13
-- aeson        >= 2    && < 2.1
-- containers   >= 0.5  && < 0.7
-- lens         >= 4.15 && < 6
-- lens-aeson   >= 1.2  && < 2
-- http-conduit >= 2.2  && < 2.4
+- base         >= 4.17  && < 5
+- text         >= 2     && < 3
+- vector       >= 0.13  && < 0.14
+- aeson        >= 2     && < 3
+- containers   >= 0.5   && < 0.7
+- lens         >= 4.15  && < 6
+- lens-aeson   >= 1.2   && < 2
+- http-conduit >= 2.2   && < 3
 
 library:
   source-dirs: src

--- a/src/Math/OEIS/Internal.hs
+++ b/src/Math/OEIS/Internal.hs
@@ -6,6 +6,7 @@ module Math.OEIS.Internal where
 import           Control.Lens        ((^?), (^?!))
 import           Control.Monad       (when)
 import           Data.Aeson.Lens
+import           Data.Aeson.Key      (fromText)
 import           Data.Aeson.Types
 import           Data.Char
 import           Data.Functor
@@ -121,7 +122,7 @@ getData result k
 
 getIntData :: Value -> T.Text -> (T.Text, Maybe OEISData)
 getIntData result k
-  = let d = result ^? key k ._Integer
+  = let d = result ^? key (fromText k) ._Integer
     in case d of
       Nothing -> (k, Nothing)
       _       ->
@@ -133,7 +134,7 @@ getIntData result k
 
 getTextData :: Value -> T.Text -> (T.Text, Maybe OEISData)
 getTextData result k
-  = let d = result ^? key k ._String
+  = let d = result ^? key (fromText k) ._String
     in case d of
       Nothing -> (k, Nothing)
       _       ->
@@ -147,11 +148,11 @@ getTextData result k
 
 getTextsData :: Value -> T.Text -> (T.Text, Maybe OEISData)
 getTextsData result k
-  = let ds  = result ^? key k . _Array
+  = let ds  = result ^? key (fromText k) . _Array
     in case ds of
       Nothing -> (k, Nothing)
       _       ->
-        let ts  = (\i -> result ^?! key k . nth i . _String) <$> [0..(len - 1)]
+        let ts  = (\i -> result ^?! key (fromText k) . nth i . _String) <$> [0..(len - 1)]
             len = fromJust $ V.length <$> ds
         in case k of
           "program" -> let prgs = parsePrograms emptyProgram [] ts

--- a/stack.yaml
+++ b/stack.yaml
@@ -1,4 +1,4 @@
-resolver: lts-14.4
+resolver: lts-20.18
 
 packages:
 - '.'

--- a/stack.yaml
+++ b/stack.yaml
@@ -1,4 +1,4 @@
-resolver: lts-20.18
+resolver: lts-22.25
 
 packages:
 - '.'

